rust/src/lib.rs

```diff
@@ -1875,7 +1875,7 @@ pub extern "C" fn zcashlc_shield_funds(
     db_data_len: usize,
     account: i32,
     xprv: *const c_char,
-    extsk: *const c_char,
+    extfvk: *const c_char,
```

Comment:
I think we could probably just remove this field entirely. `zcashlc_shield_funds` is already being used (AFAIK) uniformly with an `extsk` that corresponds to the given `account`, so we should just remove the `extsk / extfvk` parameter entirely and look up the UFVK from `db_data` (and then send to the Sapling component of it).

Reply:
How does this work with Edge, if they insist on managing all their own keys? is the `accounts` table even populated in their implementation?

Reply:
They manage the _spending keys_, not the _viewing keys_. Viewing keys have always been populated in the `accounts` table for every SDK user.

Reply:
I'm actually changing this in librustzcash now; we need to be shielding to the internal (change) address anyway. So then I'll propagate that change outward. And maybe we won't need zcashlc_derive_sapling_address_from_viewing_key then anyway.

Reply:
Opened https://github.com/zcash/librustzcash/issues/614 for removing the `extfvk` argument.

Reply:
> I think we could probably just remove this field entirely. `zcashlc_shield_funds` is already being used (AFAIK) uniformly with an `extsk` that corresponds to the given `account`, so we should just remove the `extsk / extfvk` parameter entirely and look up the UFVK from `db_data` (and then send to the Sapling component of it).

Yes, we discussed this internally and I think is a positive change. Is less error prone and more coherent. Otherwise if there's a developer error from SDK clients, they could turn the `shield_funds` into some weird `send_transparent_funds_somewhere`

Reply:
also we can control where the shielding is done.

Reply:
https://github.com/zcash/librustzcash/pull/615